src/lib/account_update.ts

```diff
@@ -1732,13 +1732,24 @@ const Authorization = {
     accountUpdate.lazyAuthorization = { ...signature, kind: 'lazy-signature' };
   },
   setProofAuthorizationKind(
-    { body }: AccountUpdate,
+    { body, id }: AccountUpdate,
     priorAccountUpdates?: AccountUpdate[]
   ) {
     body.authorizationKind.isSigned = Bool(false);
     body.authorizationKind.isProved = Bool(true);
     let hash = Circuit.witness(Field, () => {
-      priorAccountUpdates ??= zkAppProver.getData().transaction.accountUpdates;
+      let proverData = zkAppProver.getData();
+      let isProver = proverData !== undefined;
+      if (!isProver && priorAccountUpdates === undefined) {
+        throw Error(
+          'invariant violation: only the prover can call `setProofAuthorizationKind()` without passing in `priorAccountUpdates`'
```

Comment:
How can the user run into this error? I tried to reproduce it but I don't quite understand how to trigger it :sweat_smile:  

`setProofAuthorizationKind` seems only to happen when we analyze `method`s. Could we come up with a better error message that gives guidance on what the potential issues could be? The error message has some internal parameters used, which could be confusing without knowing the code of SnarkyJS. 

`priorAccountUpdates` is confusing since that mostly seems like an internal thing?

Reply:
The error is not supposed to ever happen :D I just added this as an 'assert' to make sure that we notice when the assumptions here are violated

Reply:
Got it! Do you think it would be of any value to indicate that this error should never happen and means something is really broken in a more obvious way? 

Maybe decreasing unknown error fatigue would be nice, and give a message that lets users know that it's not the fault of their zkApp?